fedimint-core/src/module/mod.rs

```diff
@@ -442,11 +441,17 @@ pub trait ModuleInit: Debug + Clone + Send + Sync + 'static {
     /// to move from the previous database version to the current version.
     const DATABASE_VERSION: DatabaseVersion;
 
-    async fn dump_database(
+    fn dump_database(
         &self,
         dbtx: &mut DatabaseTransaction<'_>,
         prefix_names: Vec<String>,
-    ) -> Box<dyn Iterator<Item = (String, Box<dyn erased_serde::Serialize + Send>)> + '_>;
+    ) -> maybe_add_send!(
+        impl Future<
```

Comment:
Yyyy... maybe not this one? :D

Reply:
more macros!

`maybe_send_future![T]`

Reply:
I heard that async in trait was still limited and something about it was unstable and recommeded to use only in private traits, etc. Is this no longer true?

I think we don't need to eliminate `#[async_trait]` altogether, we can be lazy and opportunistic about it.

Reply:
Hmmm... I think I was thinking about https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#is-it-okay-to-use---impl-trait-in-traits

> For private traits you can use -> impl Trait freely. For public traits, it's best to avoid them